File: backend/src/main/java/com/smartcampus/backend/dto/weather/AirQualitySummaryDTO.java
package com.smartcampus.backend.dto.weather;

import lombok.*;

import java.util.List;

@Getter @Setter @NoArgsConstructor @AllArgsConstructor @Builder
public class AirQualitySummaryDTO {

    private Double latitude;
    private Double longitude;

    private CurrentAirQualityDTO current;

    private List<HourlyAirQualityPointDTO> hourlyToday;

    private List<DailyAirQualityDTO> pastDaily;

    private List<DailyAirQualityDTO> forecastDaily;
}

File: frontend/app/src/main/java/com/ObservatoireCampus/mobile/model/weather/AirQualitySummaryDto.kt
package com.ObservatoireCampus.mobile.model.weather

data class AirQualitySummaryDto(
    val latitude: Double?,
    val longitude: Double?,
    val current: CurrentAirQualityDto?,
    val hourlyToday: List<HourlyAirQualityPointDto> = emptyList(),
    val pastDaily: List<DailyAirQualityDto> = emptyList(),
    val forecastDaily: List<DailyAirQualityDto> = emptyList()
)